test/parallel/test-assert-fail.js

```diff
@@ -67,5 +67,5 @@ common.expectsError(() => {
 // The stackFrameFunction should exclude the foo frame
 assert.throws(
   function foo() { assert.fail('first', 'second', 'message', '!==', foo); },
-  (err) => !/foo/m.test(err.stack)
+  (err) => !/foo(?!\/)/m.test(err.stack)
```

Comment:
This is enough to cover `/path/with/foo/node`, but it wouldn’t catch e.g. `/path/with/foo-bar/node` (or, for that matter, Windows paths that use `\`), right?

It still looks good to me as it’s strictly an improvement, but maybe there are better options.
It might be enough if the regex checks for `(whitespace)at foo `, because that’s how the function names show up?